cli/src/deploy/mod.rs

```diff
@@ -94,19 +104,36 @@ pub fn get_config() -> HashMap<String, Vec<Config>> {
   config.unwrap()
 }
 
-pub async fn post_v1(endpoint: &str, body: Value) -> Result<String, Box<dyn Error>> {
+pub async fn post_v1(endpoint: &str, body: Value) -> Result<String, PostV1Error> {
```

Comment:
I just realized maybe we can make this less verbose by splitting this into 2 functions: the first returns `Result<String, Box<dyn Error>>`, uses `?` and returns `data_str`. The other one returns `Result<String, PostV1Error>` and calls the first function

Reply:
But you mean having a function lets called it `post_v1_internal` returning `Result<String, Box<dyn Error>>` and that current post_v1 looking something like:
```
pub async fn post_v1(endpoint: &str, body: Value) -> Result<String, PostV1Error> {
  let post_v1_resp = post_v1_internal(endpoint, body).await;
  return match ...
}
```

If thats what you have in mind I think we will have the same problem as before, I will not be able to sent the different type of errors to parent fn, right?

Reply:
oh it because it has to unpack the error from internal one hmm

Reply:
correct, we can't match on a `Box<dyn Error>` and we might be able to recover from some of those hidden errors later on...